src/main/java/org/testeditor/fixture/web/json/BrowserSetupReader.java

```diff
@@ -0,0 +1,147 @@
+            List<BrowserSetting> capabilities = new ArrayList<BrowserSetting>();
+            List<BrowserSetting> options = new ArrayList<BrowserSetting>();
+            JsonObject jsonObject = browserSetupElement.getSettingsAsJsonObject();
+            setBrowserSetting(browserSetupElement, options, jsonObject, optionsString);  
+            setBrowserSetting(browserSetupElement, capabilities, jsonObject, capabilitiesString);  
+            setOsName(browserSetupElement, jsonObject);
+            setOsVersion(browserSetupElement, jsonObject);
+            setBrowserName(browserSetupElement, jsonObject);
+            setBrowserVersion(browserSetupElement, jsonObject);
+        }
+    }
+
+    private void setBrowserSetting(BrowserSetupElement browserSetupElement, List<BrowserSetting> settings, 
+            JsonObject jsonObject, String settingType) {
+        JsonElement optionsAsJSonElement = jsonObject.get(settingType);
+        if (optionsAsJSonElement != null) {
+            Set<Map.Entry<String, JsonElement>> settingEntries = optionsAsJSonElement.getAsJsonObject()
+                    .entrySet();
+            for (Map.Entry<String, JsonElement> settingEntry : settingEntries) {
+                BrowserSetting setting = createSettting(settingEntry);
+                if (setting != null) {
+                    settings.add(setting);
+                    if (settingType.equals(capabilitiesString)) {
+                        browserSetupElement.setCapabilities(settings);
+                    } else if (settingType.equals(optionsString)) {
+                        browserSetupElement.setOptions(settings);
+                    }
+                }
+            }
+        }
+    }
+    
+    private void setBrowserVersion(BrowserSetupElement browserSetupElement, JsonObject jsonObject) {
+        JsonElement browserVersion = jsonObject.get("browser-version");
+        if (browserVersion != null) {
+            browserSetupElement.setBrowserVersion(browserVersion.getAsString());
+        }
+    }
+
+    private void setBrowserName(BrowserSetupElement browserSetupElement, JsonObject jsonObject) {
+        JsonElement browser = jsonObject.get("browser");
+        if (browser != null) {
+            browserSetupElement.setBrowserName(browser.getAsString());
+        }
+    }
+
+    private void setOsVersion(BrowserSetupElement browserSetupElement, JsonObject jsonObject) {
+        JsonElement osVersion = jsonObject.get("os-version");
+        if (osVersion != null) {
+            browserSetupElement.setOsVersion(osVersion.getAsString());
+        }
+    }
+
+    private void setOsName(BrowserSetupElement browserSetupElement, JsonObject jsonObject) {
+        JsonElement os = jsonObject.get("os");
+        if (os != null) {
+            browserSetupElement.setOsName(os.getAsString());
+        }
+    }
+
+    private BrowserSetting createSettting(Entry<String, JsonElement> browserSettingEntry) {
```

Comment:
why not createBrowserSetting? is more precise and not too long, what do you think?